File: backend/src/models/apartment.model.ts
import { Schema, Model, Document } from "mongoose";
import mongoose from "mongoose";

export interface IApartment {
  apartmentCode: string;
  apartmentName: string;
  address: string;
  ownerName: string;
  ownerMobile: string;
  ownerCode: string;
  password: string;
  createdAt: Date;
  updatedAt: Date;
}

const ApartmentSchema: Schema<IApartment> = new Schema<IApartment>(
  {
    apartmentCode: {
      type: String,
      required: true,
      trim: true,
      unique: true,
    },
    apartmentName: {
      type: String,
      required: true,
      trim: true,
    },
    address: {
      type: String,
      required: true,
      trim: true,
    },
    ownerName: {
      type: String,
      required: true,
      trim: true,
    },
    ownerMobile: {
      type: String,
      required: true,
      trim: true,
    },
    ownerCode: {
      type: String,
      required: true,
      trim: true,
    },
    password: {
      type: String,
      require: true,
      trim: true,
    },
  },
  {
    timestamps: true,
  },
);

const Apartment: Model<IApartment> = mongoose.model(
  "Apartment",
  ApartmentSchema,
);

export default Apartment;

File: admin_app/lib/features/apartment/data/model/apartment_model.dart
import 'package:admin_app/features/apartment/domain/entity/apartment_entity.dart';

class ApartmentModel extends ApartmentEntity {
  ApartmentModel({
    required super.apartmentCode,
    required super.apartmentName,
    required super.ownerName,
    required super.ownerCode,
    required super.address,
    required super.password,
    required super.createdAt,
    required super.ownerMobile,
  });

  factory ApartmentModel.fromJson(Map<String, dynamic> json) => ApartmentModel(
    apartmentCode: json["apartmentCode"] ?? "",
    apartmentName: json["apartmentName"] ?? "",
    ownerName: json["apartmentName"] ?? "",
    ownerCode: json["ownerCode"] ?? "",
    address: json["ownerCode"] ?? "",
    password: json["password"] ?? "",
    createdAt: json["password"] ?? "",
    ownerMobile: json["ownerMobile"] ?? "",
  );
}
